src/core/qgstaskmanager.h

```diff
@@ -186,6 +186,14 @@ class CORE_EXPORT QgsTask : public QObject
      */
     QList< QgsMapLayer * > dependentLayers() const;
 
+    /**
+     * Blocks the current thread until the task finishes or a maximum of \a timeout milliseconds.
+     * If the \a timeout is ``-1`` the thread will be blocked forever.
+     *
+     * The result will be false if the wait timed out and true in any other case.
```

Comment:
Probably worth explicitly mentioning here that if the timeout happens the task will still be running